### DISCUSSION TITLE
BTCPay  Server 2.0 - A list of `breaking change` items

### DISCUSSION THREAD

**Discussion by pavlenex** [btcpayserver/btcpayserver#2836] [General]
The goal of this discussion is to list all the things we'd want to break in the BTCPay 2.0. Unfortunately, us trying to stay backward compatible forever bears a certain burden, and it's been discussed that we should open a list of all things that should be left behind once 2.0 launches, which at this point is most likely a year away.

- [ ] Postgres Migration (the one we supported is getting close to EOL
- [ ] TBD (Feel free to comment below)

**Comment by woutersamaey** [comment]
Fancier frontend using SPA tech (Angular, React, Vue, etc)

**Reply by ghost** [reply]
Could we add Blazor to the list? (Please don't throw any tomatoes in my direction...)

**Reply by iBobik** [reply]
Vue is already used in some parts.

**Comment by woutersamaey** [comment]
Putting an end to the 2 status system where we have `status` and `additionalStatus`.
This was created for compat with BitPay, but it sucks from a structural POV.
I guess at some point BitPay compat could be a plugin instead of core, but that may be v5 or later :laughing: ?

Also remove the `invalid` status and use proper terms for what is really meant. `invalid` is a vague term for:
- Payment not mined in time (usually 24 hours)
- Invoices manually marked as "invalid".

During my talk with @Kukks we could not find a 3rd case, so it's just this.

**Comment by Kukks** [comment]
Oh man, I don't even want to get started on this topic as it is a huge rabbit hole :)

Main thing is to make all the features atomic. This means v2 would most likely be a "reinstall and reconfigure" scenario

**Reply by pavlenex** [reply]
Get it all out man 😄

**Comment by maxmilani** [comment]
i have paid btc to open my icloud but no eny information

**Comment by JAMIEROESCH** [comment]
How can I cash out my Bitcoin with out making my own btcpay sever account

**Comment by martiankoder** [comment]
Support Hold Invoices ( Lighting Network)
